### ISSUE TITLE
Make omdict a subclass of collections.Mapping

### ISSUE THREAD

**Issue by tbodt** [gruns/orderedmultidict#10]
This would allow omdict objects to be used as GET params or form data in Requests.

**Comment by gruns** [comment]
This is a negligence of requests, not furl. Requests should not explicitly check
that `params` and `data` subclass collections.Mapping. Instead, it should check
for the existance of an `item()` method and invoke such directly. Duck typing.

That said, it's important for furl to interoperate with as many libraries as
possible, and there's no downside for omdict to subclass dict directly.

The next version of orderedmultidict, v0.7.7, will subclass dict and allow for
both of your desired requests use cases:

```python
>>> import requests
>>> from orderedmultidict import omdict
>>> o = omdict([('a', 'a'), ('b', 'b')])
>>> requests.get('http://httpbin.org/get', params=o).json().get('args')
{u'a': u'a', u'b': u'b'}
```

```python
```python
>>> import requests
>>> from orderedmultidict import omdict
>>> o = omdict([('a', 'a'), ('b', 'b')])
>>> requests.post('http://httpbin.org/post', data=o).json().get('data')
u'a=a&b=b'
```

I'll update this issue when orderedmultidict v0.7.7 is hot off the anvil.

**Comment by gruns** [comment]
Fix now live in orderedmultidict v0.7.7.


```python
>>> import requests
>>> import orderedmultidict
>>> orderedmultidict.__version__
0.7.7
>>> o = orderedmultidict.omdict([('a', 'a'), ('b', 'b')])
>>> requests.get('http://httpbin.org/get', params=o).json().get('args')
{u'a': u'a', u'b': u'b'}
>>> requests.post('http://httpbin.org/post', data=o).json().get('form')
{u'a': u'a', u'b': u'b'}
```

If you use orderedmultidict via furl, the latest furl v0.5.5 depends on the
newly minted orderedmultidict v0.7.7. Upgrade with `pip install orderedmultidict
--upgrade` or `pip install furl --upgrade`.

Thank you for bringing this issue to my attention Theodore. Don't hesitate to
let me know if there's anything else I can do for you.

**Comment by gruns** [comment]
Note: as of orderedmultidict v0.7.8, omdict objects can no longer be passed to
the `data=` argument with Requests. That is, this code

```python
>>> import requests
>>> import orderedmultidict
>>> o = orderedmultidict.omdict([('a', 'a'), ('b', 'b')])
>>> requests.post('http://httpbin.org/post', data=o).json().get('form')
```

will raise `TypeError: must be string or buffer, not set` in v0.7.8. No such
exception is raised in v0.7.7.

In v0.7.8, omdict switched to subclass `collections.MutableMapping` instead of
v0.7.7's `dict`. As learned, subclassing `dict` is a bad idea due to Python's
'magic' methods of built-in containers.


Zooming out, the fact that Requests raises an exception when `data=` is passed
an omdict object is a bug/oversight in Requests. Dictionary-like objects
shouldn't have to subclass dict itself to be a valid `data=` parameter. I
created a ticket to resolve this.

  https://github.com/kennethreitz/requests/issues/3649

Until resolved, to pass an omdict object as the `data=` parameter, either create
a native from the omdict object

```python
>>> o = orderedmultidict.omdict([('a', 'a'), ('b', 'b')])
>>> requests.post('http://httpbin.org/post', data=dict(o)).json().get('form')
```

or `urlencode()` it directly

```python
>>> from urllib import urlencode
>>> o = orderedmultidict.omdict([('a', 'a'), ('b', 'b')])
>>> requests.post('http://httpbin.org/post', data=urlencode(o)).json().get('form')
```
